### ISSUE TITLE
Bug with saving libraries/samples, without saving request

### ISSUE THREAD

**Issue by nicolazilio** [maxplanck-ie/parkour2#24]
Hello @adRn-s, 

I think I've identified a bug with adding libraries/samples. 

If I add a library or sample to a request, but do not subsequently save said request (for example by closing the request window without hitting the save button), these libraries/samples are 'lost' from the request. I can see them in the admin, but they have no request attached to them, which means that the request foreign key is added only when the request is saved. I don't think that this is how it should work, no? I mean, somebody could accidentally close the request window before saving it, and they would lose all the libraries/records that they added in the 'session'.

Any thoughts?

**Comment by nicolazilio** [comment]
I tried to solve it like this https://github.com/maxplanck-ie/parkour2/commit/33ae25a18bf59daebeca16636864f38a3283dd6a. N.B. there are differences with the parameters I pass to Ajax.request and Ext.create compared to your project.

**Comment by adRn-s** [comment]
Thanks for reporting this issue. Indeed, if the user closes the window instead of saving it, there's a data loss. I will look into your fix, would love to merge since it clearly improves the user experience without any costs 👍🏽

**Comment by nicolazilio** [comment]
The fix is not quite completely right, it works fine when you edit a request, but when adding a new one, it leads to the duplication of the request. I know why, I think :), I will let you know when I find the right solution.

**Comment by nicolazilio** [comment]
Ok, done.

The issue with auto-saving (-> creating) a new request upon hitting 'Save' in the Libraries/Samples window was that I left the mode attribute of the parent RequestWindow set to 'add'. If, upon successful creating the new request, I change this property to 'edit', as well as supply the PK of the newly created request, then when I hit 'Save' in the request window, this event edits the request, rather than saving/creating a new one (-> the duplicate).

If you want me to, I can send you a PR, so that you can check out the changes and, if you're fine with them, integrate them in your codebase.

**Comment by adRn-s** [comment]
Sure, PRs are most welcome, :) specially these that I'm sure will be appreciated by our users.

PS. I can adjust Ajax request parameters later on, when I change the models.

**Comment by nicolazilio** [comment]
Here you go https://github.com/maxplanck-ie/parkour2/pull/26
